src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/ByteArrayMemoryBlock.cs

```diff
@@ -12,7 +12,7 @@ namespace System.Reflection.Internal
     /// </summary>
     internal sealed class ByteArrayMemoryBlock : AbstractMemoryBlock
     {
-        private ByteArrayMemoryProvider _provider;
+        private ByteArrayMemoryProvider? _provider;
```

Comment:
When field is always init-ed in constructor and only nulled in the dispose it should be non null. Guidelines for this:
> DO NOT be concerned with the validity of annotations after an object has been Dispose'd. In general in .NET, using an instance after it's been disposed is a violation of its contract, and so similarly we generally (if the object isn't supposed to be used after disposal) don't consider the nullable annotations to be meaningful after disposal. This means, for example, that if a property is initialized in a ctor and is always non-null until disposal but then may return null after disposal, it should still be annotated as non-nullable.

Reply:
Addressed, thanks